### PR TITLE
Simplified edge to edge

### PR DESCRIPTION
**Pull request by android-dev-lxl**
This is the reference PR from Meghan: [Woof edge to edge](https://github.com/google-developer-training/basic-android-kotlin-compose-training-woof/pull/46)

Changes in this PR: 

- Simplified the code a bit
- Removed the code for lower SDK versions since the minimum SDK we support is 24
- Deprecated APIs have been replaced with alternate APIs
- Tested the UI on multiple emulator versions
- Fixed content padding issue

 I would appreciate it if you could take a look at these changes and let me know.

### CODE REVIEW

File: app/src/main/java/com/example/superheroes/HeroesScreen.kt [hunk 1]
```diff
@@ -63,7 +64,8 @@ import com.example.superheroes.ui.theme.SuperheroesTheme
 fun HeroesList(
     heroes: List<Hero>,
     modifier: Modifier = Modifier,
-) {
+    contentPadding: PaddingValues = PaddingValues(0.dp),
+    ) {
```
Comment (yenerm) [inline app/src/main/java/com/example/superheroes/HeroesScreen.kt:68]: i think indentation is wrong

File: app/src/main/java/com/example/superheroes/ui/theme/Theme.kt [hunk 2]
```diff
@@ -115,9 +117,7 @@ fun SuperheroesTheme(
     val view = LocalView.current
     if (!view.isInEditMode) {
         SideEffect {
-            val window = (view.context as Activity).window
-            window.statusBarColor = colorScheme.background.toArgb()
-            WindowCompat.getInsetsController(window, view).isAppearanceLightStatusBars = !darkTheme
+            setUpEdgeToEdge(view,darkTheme)
```
Comment (yenerm) [inline app/src/main/java/com/example/superheroes/ui/theme/Theme.kt:120]: leave space before darkTheme

**Review by yenerm — CHANGES_REQUESTED** [review]

**Review by MagicalMeghan — COMMENTED** [review]
LGTM for e2e in Theme.kt file

**Review by yaraki — APPROVED** [review]

File: app/src/main/java/com/example/superheroes/MainActivity.kt
Comment (johnshea) [inline app/src/main/java/com/example/superheroes/MainActivity.kt:1]: Review unused padding import.
Remove extra newline at EOF.

**Review by johnshea — APPROVED** [review]
LGTM with two quick fixes needed.

File: app/src/main/java/com/example/superheroes/HeroesScreen.kt
Comment (johnshea) [inline app/src/main/java/com/example/superheroes/HeroesScreen.kt:1]: Should the MaterialTheme shape value be used here?
Screenshot 2023-08-29 at 10 10 17 AM

Comment (johnshea) [inline app/src/main/java/com/example/superheroes/HeroesScreen.kt:1]: There are multiple occurences of `72.dp` and `8.dp`.

Can these be moved to the `dimens.xml`?

**Review by johnshea — CHANGES_REQUESTED** [review]
Update gradle wrapper to latest version. Currently `8.2`
Update `kotlinCompilerExtensionVersion` to the latest.
Update BOM to the latest.
Project build.gradle.kts - Update all plugins to the latest versions

Type.kt - ln 31 - insert blank line to push comment down
Type.kt - ln 33 & ln 41 - remove blank lines for consistency

File: app/src/main/java/com/example/superheroes/HeroesScreen.kt
Comment (android-dev-lxl) [reply app/src/main/java/com/example/superheroes/HeroesScreen.kt:1]: Yes, this was already reviewed by SMEs

Comment (android-dev-lxl) [reply app/src/main/java/com/example/superheroes/HeroesScreen.kt:1]: There is no dimens.xml for this project.

Comment (android-dev-lxl) [reply app/src/main/java/com/example/superheroes/HeroesScreen.kt:1]: Not adding for consistency.

File: build.gradle.kts [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 plugins {
-    id("com.android.application") version "8.0.2" apply false
-    id("com.android.library") version "8.0.2" apply false
-    id("org.jetbrains.kotlin.android") version "1.8.21" apply false
+    id("com.android.application") version "8.1.1" apply false
+    id("com.android.library") version "8.1.1" apply false
+    id("org.jetbrains.kotlin.android") version "1.9.0" apply false
```
Comment (johnshea) [inline build.gradle.kts:21]: nit: This isn't the latest version. Latest is `1.9.10`.

**Review by johnshea — APPROVED** [review]
One nit - build.gradle.kts - You updated to a newer version but didn't update to the latest one.

**Pull request merged**